include/safe/var.hpp

```diff
@@ -25,63 +25,108 @@ namespace safe {
+            (requirement >= detail::integral_type<U>::requirement))
+        SAFE_INLINE constexpr var(U rhs)
+            : unsafe_value_{rhs}
+        {}
+
+        template<typename U, U rhs>
+        requires (requirement.check(rhs))
+        SAFE_INLINE constexpr var(std::integral_constant<U, rhs>)
+            : unsafe_value_(rhs) // intentionally allowing narrowing conversions
+        {}
+
         SAFE_INLINE constexpr var(Var auto const & rhs)
-            : value_(rhs.unsafe_value()) // intentionally allowing narrowing conversions
+            : unsafe_value_(rhs.unsafe_value()) // intentionally allowing narrowing conversions
         {
             static_assert_assign_requirements(*this, rhs);
         }
 
         SAFE_INLINE constexpr var(Var auto && rhs)
-            : value_(rhs.unsafe_value()) // intentionally allowing narrowing conversions
+            : unsafe_value_(rhs.unsafe_value()) // intentionally allowing narrowing conversions
         {
             static_assert_assign_requirements(*this, rhs);
         }
 
         SAFE_INLINE constexpr auto operator=(Var auto const & rhs) -> var & {
             static_assert_assign_requirements(*this, rhs);
-            value_ = rhs.unsafe_value();
+            unsafe_value_ = rhs.unsafe_value();
             return *this;
         }
 
         SAFE_INLINE constexpr auto operator=(Var auto && rhs) -> var & {
             static_assert_assign_requirements(*this, rhs);
-            value_ = rhs.unsafe_value();
+            unsafe_value_ = rhs.unsafe_value();
             return *this;
         }
 
         [[nodiscard]] SAFE_INLINE constexpr T unsafe_value() const {
-            SAFE_ASSUME(requirement.check(value_));
-            return value_;
+            SAFE_ASSUME(requirement.check(unsafe_value_));
+            return unsafe_value_;
         }
 
-    private:
         using primitive_contract = detail::integral_type<T>;
 
         static_assert(rhs_must_be_subset_of_lhs<
             lhs_req<decltype(primitive_contract::requirement)>,
             rhs_req<decltype(Requirement)>
         >::value);
 
-        T value_;
+        /// @brief Do not use. Public to support use as non-type template parameter.
+        T unsafe_value_;
     };
 
+    template<typename T, typename U>
+    constexpr bool at_least_one_var = false;
+
+    template<Var T, typename U>
+    constexpr bool at_least_one_var<T, U> = true;
+
+    template<typename T, Var U>
+    constexpr bool at_least_one_var<T, U> = true;
+
+    template<Var T, Var U>
+    constexpr bool at_least_one_var<T, U> = true;
+
+
+    [[nodiscard]] SAFE_INLINE constexpr auto to_var(Var auto v) -> Var auto {
+        return v;
+    }
+
+    template<typename T, T v>
+    [[nodiscard]] SAFE_INLINE constexpr auto to_var(std::integral_constant<T, v>) -> Var auto {
+        return detail::make_constant<T, v>();
+    }
+
+    template<typename T>
+    requires (std::integral<T> || std::unsigned_integral<T>)
```

Comment:
**question**: The usual metaprogramming issue: `bool` models `std::integral` - do we want to allow `bool` here or not?